app/src/main/java/com/prac/githubrepo/main/RepoStarUpdater.kt

```diff
@@ -0,0 +1,44 @@
+package com.prac.githubrepo.main
+
+import android.view.View
+import com.prac.githubrepo.main.request.Request
+
+class RepoStarUpdater(
+    private val request: Request,
+    private val view: View
+) {
+    private var isAttachedStateListenerAdded = false
+    private val attachedStateListener = object : View.OnAttachStateChangeListener {
+        override fun onViewAttachedToWindow(view: View) {
+            startUpdatingStarState()
+        }
+
+        override fun onViewDetachedFromWindow(view: View) {
+            cancelUpdatingStarState()
+        }
+    }
+
+    private fun startUpdatingStarState() {
+        request.checkStarredState()
+    }
+
+    private fun cancelUpdatingStarState() {
+        request.cancel()
+
+        maybeRemoveListener()
+    }
+
+    fun maybeAddListener() {
+        if (isAttachedStateListenerAdded) return
+
+        isAttachedStateListenerAdded = true
+        view.addOnAttachStateChangeListener(attachedStateListener)
+    }
```

Comment:
isAttachedStateListenerAdded 라는 변수를 사옹하지 않고, 아래처럼 해보는게 어때요?
```suggestion
    fun maybeAddListener() {
        view.removeOnAttachStateChangeListener(attachedStateListener)
        view.addOnAttachStateChangeListener(attachedStateListener)
    }
```